### DISCUSSION TITLE
How to compare two fields value of different tables in Relation Manager?

### DISCUSSION THREAD

**Discussion by LucaCVG** [filamentphp/filament#10464] [Help] [answered]
HI! I'm trying to compare two values, using "Relation Manager", unfortunately without success.

The fields I'm trying to compare are in two different tables, related to each other.

I would like to be able to compare the "estimate_at_completion" field which is present in the table "EarnedValue" with the "budget_to_complete" field which is present in the table "wbe".

**i cant get the value of "wbe.budget_to_complete"**

```
                Tables\Columns\TextColumn::make('estimate_at_completion')
                    ->label('EAC')
                    ->badge()
                    ->color(fn ($state): string => floatval($state) < floatval('wbe.budget_to_complete') ? 'success' : 'danger')  
```

**Comment by leandrocfe** [comment]
Is there a wbe() relation in the model?

`->color(fn (Model $record): string => $record->estimate_at_completion < $record->wbe->budget_to_complete ? 'success' : 'danger') `

**Reply by LucaCVG** [reply]
in Wbe Model (Models\Wbe.php):

```
public function earnedvalue(): HasMany
{
return $this->hasMany(EarnedValue::class);
}
```

in EarnedValue Model (Models\EarnedValue.php):

```
public function wbe(): BelongsTo
{
return $this->belongsTo(Wbe::class , 'wbe_id');
}
```


in Wbe Resource (WbeResource.php)

```
public static function getRelations(): array
{
return [EarnedvalueRelationManager::class];
}
```

in Relation Manager of Earned Value in Wbe Resource (WbeResource\RelationManagers\EarnedvalueRelationManager.php):

```
Tables\Columns\TextColumn::make('estimate_at_completion')
->label('EAC')
->badge()
->color(fn ($state): string => floatval($state) < floatval('wbe.budget_to_complete') ? 'success' : 'danger')
->sortable(),
```

**Reply by leandrocfe** [reply, marked as answer]
`$record->wbe->budget_to_complete`
?

**Comment by LucaCVG** [comment]
Thank you! i'm here to practice and learn! thank you very much! it worked!

`->color(fn ($state, $record): string => floatval($state) < floatval($record->wbe->budget_to_complete) ? 'success' : 'danger')`
